src/metpy/remote/aws.py

```diff
@@ -0,0 +1,245 @@
+
+    def dt_from_key(self, key):
+        return datetime.strptime(key.rsplit(self.delimiter, maxsplit=1)[-1][4:19],
+                                 '%Y%m%d_%H%M%S')
+
+    def get_range(self, site, start, end):
+        for dt in date_iterator(start, end, days=1):
+            for obj in self.objects(self.build_key(site, dt, depth=4)):
+                try:
+                    if start <= self.dt_from_key(obj.key) < end:
+                        yield self._build_result(obj)
+                except ValueError:
+                    continue
+
+    def get_product(self, site, dt):
+        search_key = self.build_key(site, dt)
+        prefix = search_key.split('_')[0]
+        min_obj = min(self.objects(prefix),
+                      key=lambda o: abs((self.dt_from_key(o.key) - dt).total_seconds()))
+
+        return self._build_result(min_obj)
+
+    def _build_result(self, obj):
+        return Product(obj, lambda s: Level2File(s.file))
+
+
+@exporter.export
+class GOES16Archive(S3DataStore):
+    def __init__(self, bucket_name='noaa-goes16'):
+        super().__init__(bucket_name, delimiter='/')
+
+    def product_ids(self):
+        """Return product_ids available."""
+        return (item.rstrip(self.delimiter) for item in self.common_prefixes(''))
+
+    def build_key(self, product, dt, depth=None):
+        parts = [product, f'{dt:%Y}', f'{dt:%j}', f'{dt:%H}', f'OR_{product}']
+        return self.delimiter.join(parts[slice(0, depth)])
+
+    def _subprod_prefix(self, prefix, mode, channel):
+        subprods = set(item.rstrip('_').rsplit('-', maxsplit=1)[-1] for item in
+                       self.common_prefixes(prefix + '-', '_'))
+        if len(subprods) > 1:
+            if modes := set(item[1] for item in subprods):
+                if len(modes) == 1:
+                    mode = next(iter(modes))
+                if str(mode) in modes:
+                    prefix += f'-M{mode}'
+                else:
+                    raise ValueError(
+                        f'Need to specify a valid operating mode. Available options are '
+                        f'{", ".join(sorted(modes))}')
+            if channels := set(item[-2:] for item in subprods):
+                if len(channels) == 1:
+                    channel = next(iter(channels))
+                if str(channel) in channels:
+                    prefix += f'C{channel}'
+                elif isinstance(channel, int) and f'{channel:02d}' in channels:
+                    prefix += f'C{channel:02d}'
+                else:
+                    raise ValueError(
+                        f'Need to specify a valid channel. Available options are '
+                        f'{", ".join(sorted(channels))}')
+        return prefix
+
+    def dt_from_key(self, key):
+        start_time = key.split('_')[-3]
+        return datetime.strptime(start_time[:-1], 's%Y%j%H%M%S')
+
+    def get_product(self, product, dt, mode=None, channel=None):
```

Comment:
I think the kwarg should be "band" instead of "channel". To my knowledge the word "channel" was since the 90s more and more replaced with the synonymous word "band" e.g. AVHRR has "channels" but MODIS, VIIRS have "bands". Sometimes the word "channel" is still used today in official documents when talking more about the hardware side of the instruments. The GOES-R SERIES PRODUCT DEFINITION AND USERS’ GUIDE with 726 pages yields 25 search results for "channel" but several hundreds for "band". Oddly the GOES-ABI L2 filenames use C13, "C" for Channel though.

Reply:
Glad I'm not the only one scratching my head over when to say "band" or "channel" 😂

It does seem "band" is the preferred term in the GOES NetCDF files; some examples...
```
band_id_C01:long_name = "ABI channel 1" ;
band_id_C01:standard_name = "sensor_band_identifier" ;

band_wavelength_C01 = 0.47
band_wavelength_C01:long_name = "ABI band 1 central wavelength" ;
```

Reply:
Yeah, the "C" prefix for band/channel is the only inconsistency from the GOES side with regards to what we call it.

I'm happy to just use "band", but I think that's why "channel" comes to my mind first.